app/models/category.rb

```diff
@@ -215,6 +216,47 @@ def reset_topic_ids_cache
     Category.reset_topic_ids_cache
   end
 
+  @@category_ids_from_slugs = DistributedCache.new("category_id_from_slugs")
+
+  # Returns the category ids of the last slug in the array. The slugs array has to follow the proper category
+  # nesting hierarchy. If any of the slug in the array is invalid or if the slugs array does not follow the proper
+  # category nesting hierarchy, nil is returned.
+  #
+  # When only a single slug is provided, the category id of all the categories with that slug is returned.
+  def self.ids_from_slugs(slugs)
+    slugs = slugs.first(SiteSetting.max_category_nesting)
+
+    return [] if slugs.blank?
+
+    @@category_ids_from_slugs.defer_get_set(slugs.join(":")) do
+      params = {}
+      sql = ""
+
+      if slugs.length == 1
+        params[:slug] = slugs.first
+        sql = "SELECT id FROM categories WHERE slug = :slug"
+      else
+        slugs.each_with_index do |slug, index|
+          if index == 0
+            params[:slug_0] = slug
+            sql = "SELECT id FROM categories WHERE slug = :slug_0 AND parent_category_id IS NULL"
```

Comment:
I feel that this line is never used.

If slugs has only one element, sql from line `237` is executed

if slugs has more than one element, sql from line `246` is overriding this one on next loop iteration